tests/store/tracking/test_file_store.py

```diff
@@ -1063,6 +1063,46 @@ def test_search_runs_run_name(self):
         )
         assert [r.info.run_id for r in result] == [run1.info.run_id]
 
+    def test_search_runs_run_id(self):
+        fs = FileStore(self.test_root)
+        exp_id = fs.create_experiment("test_search_runs_run_id")
+        # Set start_time to ensure the search result is deterministic
+        run1 = fs.create_run(exp_id, user_id="user", start_time=1, tags=[], run_name="1")
+        run2 = fs.create_run(exp_id, user_id="user", start_time=2, tags=[], run_name="2")
+        run_id1 = run1.info.run_id
+        run_id2 = run2.info.run_id
+
+        result = fs.search_runs(
+            [exp_id],
+            filter_string=f"attributes.run_id IN ('{run_id1}')",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert [r.info.run_id for r in result] == [run_id1]
+
+        result = fs.search_runs(
+            [exp_id],
+            filter_string=f"attributes.run_id NOT IN ('{run_id1}')",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert [r.info.run_id for r in result] == [run_id2]
```

Comment:
@alexacole In https://github.com/mlflow/mlflow/pull/6945, we can add a test that looks like this:

```suggestion
    def test_search_runs_run_id(self):
        fs = FileStore(self.test_root)
        exp_id = fs.create_experiment("test_search_runs_run_id")
        # Set start_time to ensure the search result is deterministic
        run1 = fs.create_run(exp_id, user_id="user", start_time=1, tags=[], run_name="1")
        run2 = fs.create_run(exp_id, user_id="user", start_time=2, tags=[], run_name="2")
        run_id1 = run1.info.run_id
        run_id2 = run2.info.run_id

        result = fs.search_runs(
            [exp_id],
            filter_string=f"attributes.run_id = '{run_id1}'",
            run_view_type=ViewType.ACTIVE_ONLY,
        )
        assert [r.info.run_id for r in result] == [run_id1]

        result = fs.search_runs(
            [exp_id],
            filter_string=f"attributes.run_id != '{run_id1}'",
            run_view_type=ViewType.ACTIVE_ONLY,
        )
        assert [r.info.run_id for r in result] == [run_id2]
```